app/javascript/components/general/general_post.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const GeneralPost = ({ image, header, text, buttonText, link, positionImageRight, buttonShow, buttonClass }) => {
+  const handleButtonClick = () => window.open(link);
```

Comment:
@NRaeewal I don't think this component should control onClick behavior. It should be a dummy styled component and any parent send props to use it. I would send  onClick. as a props Check a Link component you added to this project. the same thing. We can send onClick there and Link just runs it....